avogadro/qtplugins/closecontacts/closecontacts.cpp

```diff
@@ -102,6 +104,91 @@ void CloseContacts::process(const Molecule &molecule, Rendering::GroupNode &node
         lines->addDashedLine(pos.cast<float>(), npos.cast<float>(), color, 8);
     }
   }
+
+  // Add charged atoms
+  Array<Vector3> positions;
+  Array<signed char> charges;
+  Array<Index> residues;
+  for (Index i = 0; i < molecule.atomCount(); ++i) {
+    if (molecule.formalCharge(i) != 0) {
+      if (!isAtomEnabled[i])
+        continue;
+      positions.push_back(molecule.atomPosition3d(i));
+      charges.push_back(molecule.formalCharge(i));
+      residues.push_back(Index(0) - 1);
+    }
+  }
+
+  // Add predicted charged atoms from residues
+  for (const auto &r: molecule.residues()) {
+    for (const auto &a: r.residueAtoms()) {
```

Comment:
Do you want to filter by residue? For example, this isn't important for `ALA` or `GLY` right?